### PR TITLE
feat(#185): Basic directory structure for Feedback microservice with basic query and its unit test cases.

### PR DESCRIPTION
**Pull request by anjneeksharma**
Subject: "feat/(185): Add Feedback microservice schema and basic graphql query.

# Explain the feature/fix

This ticket regarding create feedback microservice schema and add basic query like add feedback update delete.

## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Comment by anjneeksharma** [comment]
@deshmukhmayur  Please have a look on updated PR.

**Comment by riginoommen** [comment]
Please add PR description and issues related to

**Comment by anjneeksharma** [comment]
@riginoommen  can you please check it now.

File: packages/feedback-service/.env.example [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Certificate Configuration
+SSL_KEY=/etc/pki/tls/private/server.key
+SSL_CERT=/etc/pki/tls/certs/server.crt
+
+## Cookie Parser
+COOKIE_PARSER=FEEDBACK_SECRET
+
+## Authorization Token
+# AUTH_TOKEN=
```
Comment (deshmukhmayur) [inline packages/feedback-service/.env.example:25]: DO NOT edit this file. This is supposed to be an example. Whatever you need should go in a duplicate of this file called `.env`

File: packages/feedback-service/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Feedback Microservice
+=================================================
+
+One platform's server-side Feedback GraphQL microservice. This microservice will send data to Feedback-SPA that is completely optional while initializing SPA owner can check/enable feedback option and can provide basic detail like wants to use Jira/Gitlab based on SPA owner this microservice will create issue when someone will give feedback and will fetch all feedback to send to Feedback-SPA listing page.
```
Comment (deshmukhmayur) [inline packages/feedback-service/README.md:4]: Can you reword this in some simple sentences? Also, make sure whatever is written here is a part of MVP. For eg. Gitlab integration is not a part of MVP, so no need to mention that.

File: packages/feedback-service/service.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    const dbCredentials = (process.env.DB_USER && process.env.DB_PASSWORD)
+      ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@`
+      : '';
+    const dbConnection = `mongodb://${dbCredentials}${process.env.DB_PATH}/${process.env.DB_NAME}`;
+
+    mongoose.connect(dbConnection, { useNewUrlParser: true, useCreateIndex: true }).catch(console.error);
+
+    mongoose.connection.on('error', error => {
+      console.error(error);
+    });
```
Comment (deshmukhmayur) [inline packages/feedback-service/service.ts:39]: Decrease indent on these lines.

File: packages/feedback-service/src/e2e/feedback.spec.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
+import dotenv from 'dotenv';
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/e2e/feedback.spec.ts:1]: I don't see `dotenv` being used anywhere in this file. You can remove this import.

File: packages/feedback-service/src/e2e/feedback.spec.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+/* Mock */
+import mock from './mock.json';
+import path from 'path';
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/e2e/feedback.spec.ts:4]: Same for this. Not being used, so no need for the import.

File: packages/feedback-service/src/e2e/mock.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+   "message": "Define your test mocks in this file and remove this key value pair if not required"
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/e2e/mock.json:2]: This file should include the mock data for your tests. So add an object with the feedback schema and test values to be used in your unit tests.

How are your tests passing without the mock??

File: packages/feedback-service/src/schema.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import { Document, Model, model, Schema } from 'mongoose';
+import { IFeedback } from "./types";
+
+export const FeedbackSchema: Schema = new Schema({
+  description: String,
+  experience: String,
+  iid: String,
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:7]: Rename this field to `ticketNumber` or `ticketID`. `iid` is very ambiguous.

File: packages/feedback-service/src/schema.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  spa: String,                
+  portalFeedback: Boolean,
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:9]: This field is not needed in the new Framework.

File: packages/feedback-service/src/resolver.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        .catch(err => err);
+    },
+    getFeedback(root: any, args: any, ctx: any) {
+      return Feedback.findById(args._id)
+        .then(response => response)
+        .catch(err => err);
+    }
+  },
+  Mutation: {
+    // mutations
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:22]: Unnecessary comment

File: packages/feedback-service/src/resolver.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    addFeedback(root: any, args: any, ctx: any) {
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:23]: Refactor this resolver to remove the email logic. It is not needed, as JIRA will take care of that.
And even after that, yo can use the Notifications Microservice to send out the emails.

File: packages/feedback-service/src/helpers.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    host: "smtp.corp.redhat.com",
+    port: 587,
+    secure: false,
+    tls: {
+      // do not fail on invalid certs
+      rejectUnauthorized: false
+    },
+    logger: true,
+    debug: false // include SMTP traffic in the logs
+  });
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/helpers.ts:16]: The nodemailer is not needed in the framework. Most of the email communication will be handled by JIRA, and if we need custom emails even after that, you can use the Notifications Microservice to send out the emails.

File: packages/feedback-service/src/resolver.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
+import { Feedback } from './schema';
+import { getTransporter } from './helpers';
+  
+const transporter = getTransporter;
+
+export const FeedbackResolver = {
+  Query: {
+    //queries
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:8]: Unnecessary comment

File: packages/feedback-service/src/typedef.graphql [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  experience: String
+  module: String
+  portalFeedback: Boolean
+  timestamp: TimestampInput
+  title: String
+  feedbackType: String
+}
+
+type Query {
+  # Query Definitions
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:39]: This comment should describe the query/mutation on the line below it.
So here it should describe what listFeedback does.

This is helpful when using GraphQL query builders, as this is auto-documented.

Add the same for all the queries and mutations.

File: packages/feedback-service/src/typedef.graphql [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+  listFeedback: [FeedbackType]
+  getFeedback(_id: String): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:41]: Add one more query to getFeedback by any property. `getFeedbackBy`

Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:41]: Avoid using `_id` in any interface or user facing things. Just `id` is clear and easy to understand and use.

```suggestion
  getFeedback(id: String): FeedbackType
```

File: packages/feedback-service/src/typedef.graphql [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+type Mutation {
+  # Mutation Definitions
+  addFeedback(input: FeedbackInput): FeedbackType
+  updateFeedback(input: FeedbackInput): FeedbackType
+  deleteFeedback(_id: String!): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:48]: Just `id`

```suggestion
  deleteFeedback(id: String!): FeedbackType
```

File: packages/feedback-service/src/types.d.ts [hunk 16]
```diff
@@ -0,0 +1,37 @@
+declare module '*.graphql';declare module '*.json';
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/types.d.ts:1]: Each declaration should be on a separate line.

```suggestion
declare module '*.graphql';
declare module '*.json';
```

File: packages/feedback-service/.env.example [hunk 17]
```diff
@@ -0,0 +1,25 @@
+## Uncomment the lines as needed
+
+NAME=local
+NODE_ENV=local
+PRODUCTION=false
+
+## Server URLs
+API=https://localhost:8080
+GRAPHQL_API=https://localhost:8080/playground
```
Comment (deshmukhmayur) [inline packages/feedback-service/.env.example:9]: This url is invalid as we're not using GraphQL playground anymore.

File: packages/feedback-service/.env.example (hunk 1 above)
Comment (anjneeksharma) [reply packages/feedback-service/.env.example:25]: Okay, this file was created automatically when i created directory structure after that i thought there is not use of having this file so deleted this somehow it generated again.
Noted i will not update this.

File: packages/feedback-service/README.md (hunk 2 above)
Comment (anjneeksharma) [reply packages/feedback-service/README.md:4]: Okay.

File: packages/feedback-service/.env.example (hunk 17 above)
Comment (anjneeksharma) [reply packages/feedback-service/.env.example:9]: yes confirmed with Rigin so replacing with.
GRAPHQL_API=https://localhost:8080/graphql

File: packages/feedback-service/src/typedef.graphql (hunk 14 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/typedef.graphql:41]: So you mean user can pass single property or can pass all the property  including _id ?

**Comment by anjneeksharma** [comment]
@deshmukhmayur  have another look.

File: packages/feedback-service/README.md [hunk 18]
```diff
@@ -1,7 +1,7 @@
 Feedback Microservice
 =================================================
 
-One platform's server-side Feedback GraphQL microservice. This microservice will send data to Feedback-SPA that is completely optional while initializing SPA owner can check/enable feedback option and can provide basic detail like wants to use Jira/Gitlab based on SPA owner this microservice will create issue when someone will give feedback and will fetch all feedback to send to Feedback-SPA listing page.  
+One platform's server-side Feedback GraphQL microservice. This microservice will allow us to talk to feedback database and can perform operation like  addfeedback,updatefeedback,deletefeedback,listfeedback.
```
Comment (sayak-sarkar) [inline packages/feedback-service/README.md:4]:
```suggestion
One platform's server-side Feedback GraphQL microservice. This microservice will allow users to talk to the feedback database and can perform operations like  addfeedback, updatefeedback, deletefeedback & listfeedback.
```

File: packages/feedback-service/src/typedef.graphql (hunk 14 above)
Comment (deshmukhmayur) [reply packages/feedback-service/src/typedef.graphql]: You can pass a partial object of type `FeedbackInput`. And any properties that match the input object will be returned.

File: packages/feedback-service/src/resolver.ts [hunk 19]
```diff
@@ -1,70 +1,32 @@
...
         .catch(err => err);
     },
     getFeedback(root: any, args: any, ctx: any) {
-      return Feedback.findById(args._id)
+      return Feedback.findById(args.id)
         .then(response => response)
         .catch(err => err);
+    },
+    getFeedbackBy(root: any,  args : any, ctx: any){
+      const searchBy = _.pick(args, ["input"]);
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:18]: The `_.pick` is to pick out selected properties from an object. What you're doing here has no effect.
You need to pick out the properties from the `args.input` object. And use those properties in the find.

File: packages/feedback-service/src/resolver.ts [hunk 20]
```diff
@@ -1,70 +1,32 @@
...
+      return Feedback.find(searchBy.input)
+          .then(response=>response)
+          .catch(err=>err);
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:21]:
```suggestion
      return Feedback.find(searchBy.input).exec();
```

File: packages/feedback-service/src/typedef.graphql [hunk 21]
```diff
@@ -26,27 +38,28 @@ type FeedbackType {
...
   feedbackType: String
 }
 
 type Query {
-  # Query Definitions
+  # Fetches all Feedbacks
   listFeedback: [FeedbackType]
-  getFeedback(_id: String): FeedbackType
+  # Fetches specific Feedback by its id
+  getFeedback(id: String): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:53]: You can use `!` to mark the parameters as required.

```suggestion
  getFeedback(id: String!): FeedbackType
```

Comment (deshmukhmayur) [reply packages/feedback-service/src/typedef.graphql:53]: Do the same for `getFeedbackBy` and the mutations.

File: packages/feedback-service/src/types.d.ts [hunk 22]
```diff
@@ -1,37 +1,26 @@
-declare module '*.graphql';declare module '*.json';
+declare module '*.graphql';
+declare module '*.json';
 
-export interface IUserTeam {
-    name: string;
-    access: number;
-    primary: Boolean;
-  }
+declare const _;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/types.d.ts:4]: No need for this declaration for the `_` type. Install the type definitions for lodash: `@types/lodash`

File: packages/feedback-service/.env.example [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+DB_NAME=db_name
+# DB_USER=
+# DB_PASSWORD=
+
+## Certificate Configuration
+SSL_KEY=/etc/pki/tls/private/server.key
+SSL_CERT=/etc/pki/tls/certs/server.crt
+
+## Cookie Parser
+COOKIE_PARSER=FEEDBACK_SECRET
```
Comment (deshmukhmayur) [inline packages/feedback-service/.env.example:22]:
```suggestion
COOKIE_PARSER=
```

File: packages/feedback-service/package.json [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+  "scripts": {
+    "test": "cross-env NODE_ENV=test DB_PATH=localhost DB_NAME=feedback-db-test jest --forceExit",
+    "build:dev": "webpack --config webpack.dev.js",
+    "start:dev": "webpack --watch --config webpack.dev.js",
+    "build": "webpack --config webpack.prod.js",
+    "start": "webpack --watch --config webpack.prod.js"
+  },
+  "author": {
+    "name": "Anjnee K Sharma",
+    "email": "anjnee.k.sharma@gmail.com"
```
Comment (deshmukhmayur) [inline packages/feedback-service/package.json:34]: Use your redhat email

File: packages/feedback-service/src/e2e/mock.json [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+        "name": "Anjnee K"
+      },
+      "modifiedAt": "2019-05-30T11:22:48.706Z",
+      "modifiedBy":{
+        "kerberosID": "anjsharm",
+        "name": "Yo Yo"
+      } 
+  },
+    "title": "second record",
+    "feedbackType": "Feedback"
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/e2e/mock.json:20]: Decrease indent, and move to the top after "_id".

**Comment by deshmukhmayur** [comment]
@anjneeksharma Also, please add proper descriptive titles for your PRs.

File: packages/feedback-service/src/resolver.ts (hunk 20 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts:21]: if yo can you help me in understand why .then .catch are useless are i seen other developer also implement the same.

File: packages/feedback-service/src/e2e/mock.json [hunk 26]
```diff
@@ -1,4 +1,21 @@
-{
-   "message": "Define your test mocks in this file and remove this key value pair if not required"
+{ 
+   "_id": "5e9dc1b281c18d6de9627e35",
+   "description": "this is frist document",
+   "ticketID": "200",
+   "experience": "its good",
+   "spa": "2nd test feedback",
```
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/src/e2e/mock.json:6]: you mean in place of "2nd test feedback" there should ID if yes then which ID ?
do i need to change schema or resolver for it ?

File: packages/feedback-service/package.json [hunk 27]
```diff
@@ -23,7 +23,7 @@
     "dist"
   ],
   "scripts": {
-    "test": "jest --forceExit",
+    "test": "cross-env NODE_ENV=test DB_PATH=localhost DB_NAME=feedback-db-test jest --forceExit",
```
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/package.json:26]: I tried this having test.env  file inside e2e/test.env  and also on similar path of .env but was not able to read value from this file.
Do i need any other extra setting like webpack or something  ?

File: packages/feedback-service/src/e2e/mock.json (hunk 26 above)
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/src/e2e/mock.json]: spa: { type: Schema.Types.ObjectId, ref: "?" }   then while listing in resolver need to populate from the collection which passed in ref

[the remark this answers is not in the document]
Comment (deshmukhmayur) [reply packages/feedback-service/src/e2e/mock.json]: You can keep the type as String in the schema. String can be typecasted to ObjectId later, when we have the SPA db.

File: packages/feedback-service/src/resolver.ts (hunk 20 above)
Comment (deshmukhmayur) [reply packages/feedback-service/src/resolver.ts]: The `.then` and `.catch` are used to handle certain cases. For eg. then is used when you want to perform certain operations on the data, and catch is used when you want to handle the error in some other way.
Here, you were just return the arguments as it is, so it did not have any effect.
Instead, graphql expects the resolvers to return a promise, and it can take care of the rest, so just `.find().exec()` is sufficient.

File: packages/feedback-service/src/e2e/mock.json [hunk 28]
```diff
@@ -1,21 +1,22 @@
...
+      "kerberosID": "anjsharm",
+      "name": "Anjnee K"
+    },
+    "modifiedAt": "2019-05-30T11:22:48.706Z",
+    "modifiedBy":{
+      "kerberosID": "anjsharm",
+      "name": "Yo Yo"
+    } 
+}
+
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/e2e/mock.json:21]:
```suggestion
}
```

File: packages/feedback-service/package.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (deshmukhmayur) [reply packages/feedback-service/package.json:26]: The `.test.env` needs to be in the `src/e2e/` directory. Check the webpack config. When testing, it automatically injects the variables from `.test.env` into the environment.

File: packages/feedback-service/src/typedef.graphql [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+  listFeedback: [FeedbackType]
+  # Fetches specific Feedback by its id
+  getFeedback(id: String!): FeedbackType
+  #Fetches Feedbacks by any property
+  getFeedbackBy(input: FeedbackInput!):[FeedbackType]
+}
+
+type Mutation {
+  # Add a new Feedback
+  addFeedback(input: FeedbackInput): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:60]:
```suggestion
  addFeedback(input: FeedbackInput!): FeedbackType
```

File: packages/feedback-service/src/typedef.graphql [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+  # Update Feedback based on input
+  updateFeedback(input: FeedbackInput): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:62]:
```suggestion
  updateFeedback(input: FeedbackInput!): FeedbackType
```

File: packages/feedback-service/src/resolver.ts (hunk 20 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts]: Got it.

File: packages/feedback-service/service.ts [hunk 31]
```diff
@@ -37,27 +37,13 @@ mongoose.connection.on('error', error => {
...
-    locations: error.locations,
-    stack: error.stack ? error.stack.split('\n') : [],
-    path: error.path,
-  }),
-  extensions
-});
+  schema: buildFederatedSchema( [ {
+    typeDefs: gqlSchema,
+    resolvers: resolver
+  }])});
```
Comment (deshmukhmayur) [inline packages/feedback-service/service.ts:45]: Why did you remove the rest of the options?
The `subscriptions` and `formatError`?

File: packages/feedback-service/package.json [hunk 32]
```diff
@@ -36,12 +36,15 @@
...
   "license": "MIT",
   "dependencies": {
+    "@apollo/federation": "^0.14.1",
     "apollo-log": "^0.1.0",
     "apollo-server-express": "^2.10.1",
     "body-parser": "^1.19.0",
     "clean-webpack-plugin": "^3.0.0",
     "cookie-parser": "^1.4.4",
     "cors": "^2.8.5",
+    "create-jira-ticket-api": "^1.0.1",
```
Comment (deshmukhmayur) [inline packages/feedback-service/package.json:46]: Use the `jira-client` package instead of this one.
https://www.npmjs.com/package/jira-client
It has more downloads and is maintained. 

`create-jira-ticket-api` has not been updated since last 3 years.

File: packages/feedback-service/package.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/package.json:26]: "build:dev": "webpack --config webpack.dev.js", === >module.exports = merge(webpack.common.js, { 
thats how its reaching up to comman webpack in case of unit testing command-webpack not being called i tried it.

File: packages/feedback-service/service.ts (hunk 31 above)
Comment (anjneeksharma) [reply packages/feedback-service/service.ts:45]: As i seen in rigin email screenshot it was not there, i will revert them.

File: packages/feedback-service/package.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (deshmukhmayur) [reply packages/feedback-service/package.json:26]: Actually you're right @anjneeksharma 

@riginoommen Why's the .test.env configured inside the webpack?
It needs to be configured in the test files.
Also, have you tested it on user-service? Was the test.env working there?

[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/feedback-service/package.json:26]: Yes its working for me

[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/package.json:26]: @riginoommen  when i run my npm test webpack.comman.js is not being called then how it will set envPath ?  
webpack.comman.js is being called from webpack.dev.js and and  webpack.prod.js and those prod and dev webpack will be called only when npm run start or npm run start:dev is being run not in case of npm test.

File: packages/feedback-service/service.ts (hunk 31 above)
Comment (anjneeksharma) [reply packages/feedback-service/service.ts:45]: @riginoommen  can you please confirm here as i can see in your email you sent there is no subscriptions and formatError ?

File: packages/feedback-service/package.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/package.json:26]: @riginoommen  i think why its working for you because you execute 2 command together npm run start:dev && npm test if you will execute only one command npm test it will fail for you also if i am not wrong.

[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-service/package.json:26]: @deshmukhmayur  i gone through multiple article and youtube video what i found 
Environment variable can be set in 3 place:-
  1 project.json
  2 jest.config.ts 
  3 inside test file itself in beforeAll() function 

we can keep our env variable in src/e2e/.test.env file as well but in that case we need to use npm test --env=<environment>  meaning   **npm test --env src/e2e** this will also work fine.

Please let me know which one i should use.

File: packages/feedback-service/package.json (hunk 32 above)
Comment (anjneeksharma) [reply packages/feedback-service/package.json:46]: Removed.

File: packages/feedback-service/src/resolver.ts (hunk 19 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts:18]: Actually _.pick is not required here i tested return Feedback.find(args.input).exec(); like this mongoDB take care of it and i also printed filed there which ever fileds i am passing from browser only those field are there.
so i am removing _.pick from here.

File: packages/feedback-service/service.ts (hunk 31 above)
Comment (deshmukhmayur) [reply packages/feedback-service/service.ts:45]: @anjneeksharma That email was an example, so he only mentioned the part that needed to be changed. Read the email carefully. He specifically mentioned not to touch the other config files.
Also, the subscriptions and formatError is not related to the schemas.

Comment (deshmukhmayur) [reply packages/feedback-service/service.ts:45]: Also, the subscriptionsHandlers are being installed on line 63, to if you remove this, that line will probably throw an error.

**Comment by anjneeksharma** [comment]
@sayak-sarkar  please have a look.

**Comment by anjneeksharma** [comment]
@riginoommen  Please have a look.
